Init commit

KozlovskiyD opened the pull request:
Toast all
Домашная работа часть 3.5
Домашная работа спринт 9.1
Домашняя работа спринт 10

Dmitr80 committed:
Init commit
Toast all
Домашная работа часть 3.5
Домашная работа спринт 9.1
Домашняя работа спринт 10

AleksandrIlinskii commented on .idea/render.experimental.xml:
```diff
@@ -0,0 +1,6 @@
+  <component name="RenderSettings">
+    <option name="showDecorations" value="true" />
+  </component>
+</project>
```
Этот файл и файл выше - системные файлы, их лучше не пушить, потому что они могут затруднить запуск твоего проекта у коллеги, который будет себе клонировать проект

AleksandrIlinskii commented on app/build.gradle:
```diff
@@ -9,7 +9,7 @@ android {
     defaultConfig {
         applicationId "com.practicum.playlistmaker3"
-        minSdk 33
+        minSdk 31
```
minSdk показывает, какая минимальная версия android требуется для запуска проекта. Сейчас это 12, лучше уменьшить число для большего охвата аудитории

AleksandrIlinskii commented on app/build.gradle:
```diff
@@ -34,11 +34,15 @@ android {
 
-    implementation 'androidx.core:core-ktx:1.7.0'
-    implementation 'androidx.appcompat:appcompat:1.6.1'
+    implementation 'androidx.core:core-ktx:1.9.0'
```
Поднимать версии библиотек полезно, но будь аккуратнее. Некоторые несовместимы и могут сломать билд

AleksandrIlinskii commented on app/src/main/java/com/practicum/playlistmaker3/Keyboard.kt:
```diff
@@ -0,0 +1,10 @@
+import android.view.View
+import android.view.inputmethod.InputMethodManager
+
+fun Context.hideTheKeyboard(currentView: View) {
```
Название файла не отображает содержимое, сейчас Keyboard говорит о том, что внутри что-то связано с клавиатурой, а тут extension функция. Лучше переименовать в Extension

AleksandrIlinskii commented on app/src/main/java/com/practicum/playlistmaker3/SearchActivity.kt:
```diff
@@ -45,6 +44,12 @@ class SearchActivity : AppCompatActivity() {
+        val trackListAdapter = TrackListAdapter(
+            List(5){ Tract( trackNames[it], artistNames[it], trackTimes[it], artworkUrls[it] ) })
+
+        val rvTractList = findViewById<RecyclerView>(R.id.TrackList)
```
Опечатка - rvTrackList

AleksandrIlinskii commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -19,55 +20,65 @@
     </FrameLayout>
 
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/TrackList"
```
id пишется с маленькой буквы

AleksandrIlinskii commented on app/src/main/java/com/practicum/playlistmaker3/Treck.kt:
```diff
@@ -0,0 +1,47 @@
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.resource.bitmap.RoundedCorners
+
+data class Tract(val trackName: String, val artistName: String, val trackTime: String, val artworkUrl100: String)
```
Дата класс надо вынести в отдельный файл

AleksandrIlinskii commented on app/src/main/java/com/practicum/playlistmaker3/Treck.kt:
```diff
@@ -0,0 +1,47 @@
+    private val tvTractTime: TextView = itemView.findViewById(R.id.tvTrackTime)
+
+    fun bind(model: Tract) {
+        model.trackName.also { tvTractName.text = it }
```
also излишне, просто лучше написать tvTractName.text = model.trackName
С остальными ниже аналогично

AleksandrIlinskii commented on app/src/main/java/com/practicum/playlistmaker3/Treck.kt:
```diff
@@ -0,0 +1,47 @@
+    }
+}
+
+class TrackListAdapter (private val tracks: List<Tract>): RecyclerView.Adapter<TrackListViewHolder>() {
```
Адаптер надо вынести в отдельный файл

AleksandrIlinskii commented on app/src/main/java/com/practicum/playlistmaker3/SearchActivity.kt:
```diff
@@ -71,8 +68,23 @@ class SearchActivity : AppCompatActivity() {
 
+    companion object {
+        const val SAVE_TEXT = "SAVE_TEXT"
+        val trackNames = listOf("Smells Like Teen Spirit", "Billie Jean", "Stayin' Alive", "Whole Lotta Love", "Sweet Child O'Mine")
```
Лучше сделать вместо 4-х листов с данными 1 лист List<Track>

AleksandrIlinskii commented on app/src/main/java/com/practicum/playlistmaker3/SettingsActivity.kt:
```diff
@@ -19,7 +18,7 @@ class SettingsActivity : AppCompatActivity() {
         }
 
-        val switchNight = findViewById<Switch>(R.id.switch1)
+        val switchNight = this.findViewById<Switch>(/* id = */ R.id.switch1)
```
именование аргументов в java функциях не работает, поэтому студия и делает как комментарий. Он излишен, его лучше удалить, как и в других местах проекта

AleksandrIlinskii commented on app/src/main/java/com/practicum/playlistmaker3/SettingsActivity.kt:
```diff
@@ -28,23 +27,25 @@ class SettingsActivity : AppCompatActivity() {
-             writeToIntent.putExtra(Intent.EXTRA_TEXT, massage)
-             startActivity(writeToIntent)
+             Intent(Intent.ACTION_SENDTO).apply {
+                 val massage = this@SettingsActivity.getString (R.string.massage)
```
Если имелось ввиду сообщение, то это message

AleksandrIlinskii commented on app/src/main/java/com/practicum/playlistmaker3/SettingsActivity.kt:
```diff
@@ -28,23 +27,25 @@ class SettingsActivity : AppCompatActivity() {
 
         val buttonShared = findViewById<ImageView>(R.id.write_to)
```
Лучше назвать buttonShare, Shared - это прошедшее время глагола share - поделиться

AleksandrIlinskii commented on app/src/main/java/com/practicum/playlistmaker3/Treck.kt:
```diff
@@ -0,0 +1,47 @@
+        Glide
+            .with(itemView)
+            .load(model.artworkUrl100)
+            .placeholder(R.drawable.ic_launcher_background)
```
Вставь какую-нибудь картинку. Смотреть на зеленый фон - не лучшее решение для UI

AleksandrIlinskii commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -12,34 +12,40 @@
     <TextView
         style="@style/headingText"
         android:layout_width="wrap_content"
+        android:layout_height = "wrap_content"
```
Пройдись в файле автоформатирование, Code - Refactoring code. Он расставит отступы согласно правилам языка

AleksandrIlinskii commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -19,55 +20,65 @@
+        android:layout_width = "wrap_content"
+        android:layout_height = "wrap_content"
+        android:layout_marginTop = "@dimen/top_padding"
+        android:layout_marginStart = "40dp"
```
Все значения, если они повторяются - выносим в ресурсы

AleksandrIlinskii commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -19,55 +20,65 @@
 
 <FrameLayout
     android:layout_width="match_parent"
     android:layout_height="wrap_content">
```
Из-за того, что тут стоит не на всю доступную часть экрана, а на его часть - то получается вот так
![image](https://user-images.githubusercontent.com/129396632/230732586-73b48a44-4afe-4c77-9127-d059e8633039.png)

Dmitr80 committed:
Init commit
Toast all
Домашная работа часть 3.5
Домашная работа спринт 9.1
Домашняя работа спринт 10.1

AleksandrIlinskii commented on app/src/main/java/com/practicum/playlistmaker3/DataTrack.kt:
```diff
@@ -0,0 +1,3 @@
+package com.practicum.playlistmaker3
+
+data class Tract(val trackName: String, val artistName: String, val trackTime: String, val artworkUrl100: String)
```
Файл лучше назвать Track, так как внутри 1 дата класс. Проверь пожалуйста написано Track, там опечатка

AleksandrIlinskii commented on app/src/main/java/com/practicum/playlistmaker3/SearchActivity.kt:
```diff
@@ -73,10 +73,10 @@ class SearchActivity : AppCompatActivity() {
+        val trackList = listOf(
+            "Smells Like Teen Spirit", "Billie Jean", "Stayin' Alive", "Whole Lotta Love", "Sweet Child O'Mine",
+            "Nirvana", "Michael Jackson", "Bee Gees", "Led Zeppelin", "Guns N' Roses",
+            "5:01", "4:35", "4:10", "5:33", "5:03",
```
Необходимо создать 1 лист Track, а сейчас это 4 листа String
